src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.IncrementalAnalyzerProcessor.cs

```diff
@@ -401,7 +401,7 @@ private static bool CrashUnlessCanceled(AggregateException aggregate)
                         return true;
                     }
 
-                    FatalError.Report(flattened);
+                    FatalError.ReportWithoutCrashAndPropagate(flattened);
```

Comment:
💡 Containing method name is no longer correct

Reply:
Good catch.

Reply:
I think this code can be cleaned up further:

```C#
try
{
    return await funcAsync(value, cancellationToken).ConfigureAwait(false);
}
catch (OperationCanceledException)
{
    return null;
}
catch (AggregateException e) when (ReportWithoutCrashUnlessAllCanceledAndPropagate(e))
{
    return null;
}
catch (Exception e) when (FatalError.ReportWithoutCrashAndPropagate(e))
{
    // TODO: manage bad workers like what code actions does now
    throw ExceptionUtilities.Unreachable;
}
```

As you pointed out the other day awaiting a task should unwrap AggregateException, correct? So we shouldn't need to handle it ourselves.

Reply:
Now, it's not clear where the AggregateException is actually unwrapped. Remmeber the try-catch I had around Task.WhenAll?

Reply:
> Now, it's not clear where the AggregateException is actually unwrapped

It's unwrapped when `TaskAwaiter<TResult>.GetResult()` is called by the continuation.

https://github.com/dotnet/runtime/blob/1beb1c635a2961d96d56d52025ee0fa7322933bb/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/TaskAwaiter.cs#L175-L181